Firebase/Storage/FIRStorage.m

```diff
@@ -126,15 +139,20 @@ + (instancetype)storageForApp:(FIRApp *)app URL:(NSString *)url {
     bucket = path.bucket;
   }
 
-  return [[self alloc] initWithApp:app bucket:bucket];
+  // Retrieve the instance provider from the app's container to inject dependencies as needed.
+  id<FIRStorageMultiBucketProvider> provider =
+      FIR_COMPONENT(FIRStorageMultiBucketProvider, app.container);
+  return [provider storageForBucket:bucket];
```

Comment:
Should the return type of the function be FIRStorageComponent instead of instancetype?

Reply:
Nope - we're returning the results from `storageForBucket:` which returns a `FIRStorage` instance (`instancetype` since we're in the `FIRStorage` class).